File: web/src/api/expertSearch.js
import service from '@/utils/request'

// @Tags ExpertSearch
// @Summary 专家综合推荐排序检索
// @Router /expertDatabase/search [get]
export const searchExpert = (params) => {
  return service({ url: '/expertDatabase/search', method: 'get', params })
}

// @Tags ExpertSearch
// @Summary 手动重算专家得分
// @Router /expertDatabase/recomputeScore [post]
export const recomputeExpertScore = (data) => {
  return service({ url: '/expertDatabase/recomputeScore', method: 'post', data })
}

// @Tags ExpertSearch
// @Summary 导出当前检索结果
// @Router /expertDatabase/exportSearchResults [get]
export const exportSearchResults = (params) => {
  return service({ url: '/expertDatabase/exportSearchResults', method: 'get', params, responseType: 'blob' })
}

// @Tags ExpertSearch
// @Summary 分项得分是否有数据（全库是否至少一人非零），用来决定要不要显示对应的列/卡片
// @Router /expertDatabase/scoreColumnAvailability [get]
export const getScoreColumnAvailability = () => {
  return service({ url: '/expertDatabase/scoreColumnAvailability', method: 'get' })
}

File: server/router/ExpertDatabase/expert_search.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertSearchRouter struct{}

// InitExpertSearchRouter 初始化 专家综合推荐排序检索 路由信息
func (s *ExpertSearchRouter) InitExpertSearchRouter(Router *gin.RouterGroup) {
	expertDatabaseRouter := Router.Group("expertDatabase").Use(middleware.OperationRecord())
	expertDatabaseRouterWithoutRecord := Router.Group("expertDatabase")
	expertSearchApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertSearchApi
	{
		expertDatabaseRouter.POST("recomputeScore", expertSearchApi.RecomputeExpertScore) // 手动重算专家得分
	}
	{
		expertDatabaseRouterWithoutRecord.GET("search", expertSearchApi.SearchExpert)                                       // 综合推荐排序检索
		expertDatabaseRouterWithoutRecord.GET("exportSearchResults", expertSearchApi.ExportSearchResults)                   // 导出检索结果
		expertDatabaseRouterWithoutRecord.GET("scoreColumnAvailability", expertSearchApi.GetScoreColumnAvailability)        // 分项得分是否有数据
	}
}
